### PR TITLE
[MIG]12.0 Migrate sale_order_lot_selection

### PR DESCRIPTION
**Pull request by PCatinean**

### CODE REVIEW

File: sale_order_lot_selection/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# © 2015 Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+{
+    'name': 'Sale Order Lot Selection',
+    'version': '12.0.0.0.0',
+    'category': 'Sales Management',
+    'author': "Odoo Community Association (OCA), Agile Business Group",
+    'website': 'http://www.agilebg.com',
```
Comment (nikul-serpentcs) [inline sale_order_lot_selection/__manifest__.py]:
```suggestion
'website': 'https://github.com/OCA/sale-workflow',
```

File: sale_order_lot_selection/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# © 2015 Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+{
+    'name': 'Sale Order Lot Selection',
+    'version': '12.0.0.0.0',
```
Comment (nikul-serpentcs) [inline sale_order_lot_selection/__manifest__.py]:
```suggestion
    'version': '12.0.1.0.0',
```

File: sale_order_lot_selection/__init__.py [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# © 2015 Agile Business Group
```
Comment (nikul-serpentcs) [inline sale_order_lot_selection/__init__.py]: @PCatinean no need to copyright in `init` file, Remove it

File: sale_order_lot_selection/models/__init__.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#########################################################################
```
Comment (nikul-serpentcs) [inline sale_order_lot_selection/models/__init__.py]: same here

File: sale_order_lot_selection/models/sale.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# © 2015 Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo import api, fields, models, _
+from odoo.exceptions import Warning
```
Comment (nikul-serpentcs) [inline sale_order_lot_selection/models/sale.py]: use UserError here

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Improve Code

**Comment by PCatinean** [comment]
I guess there's a small issue with the clabot for amended commits?

**Comment by PCatinean** [comment]
Are there any other changes needed?

File: sale_order_lot_selection/models/procurement.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
+from odoo import api, fields, models
```
Comment (nicomacr) [inline sale_order_lot_selection/models/procurement.py]: This file it's not necessary now , procurement.order not longer available in 12.0

File: sale_order_lot_selection/models/stock.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        if self._context.get('sol_lot_id'):
+            lot_id = self.sale_line_id.lot_id
+        return super(StockMove, self)._update_reserved_quantity(
+            need, available_quantity, location_id, lot_id=lot_id,
+            package_id=package_id, owner_id=owner_id, strict=strict)
+
+    def _prepare_move_line_vals(self, quantity=None, reserved_quant=None):
+        vals = super(StockMove, self)._prepare_move_line_vals(
+            quantity=quantity, reserved_quant=reserved_quant)
+        if reserved_quant:
```
Comment (nicomacr) [inline sale_order_lot_selection/models/stock.py]: here must be validate if the lot it's setting in the sale order line, becase it's set a lot in the move then with this set always false.

`if reserved_quant and self.sale_line_id.lot_id:`

**Review by nicomacr — COMMENTED** [review]
Hi @PCatinean ,

I've made a review for this PR.
Put some comment below

**Comment by mohamedhagag** [comment]
Just did functional test on runbot  and working fine, should we merge ?

**Review by wpichler — APPROVED** [review]

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1: 

Please Fix other comments https://github.com/OCA/sale-workflow/pull/785#discussion_r271783360 https://github.com/OCA/sale-workflow/pull/785#discussion_r271784120

**Review by bodedra — APPROVED** [review]
@PCatinean Functional review. LGTM. :+1:

Non blocking comment: Would you please add readme folder and needed files in it?

**Comment by nikul-serpentcs** [comment]
@bodedra  https://github.com/OCA/sale-workflow/pull/785#discussion_r271783360 this comment remains because of `procurement.order` not much support in v12.
So, we need to change file and methods too

**Comment by rousseldenis** [comment]
@PCatinean Is this ready ?

**Comment by PCatinean** [comment]
Requested changes applied and also merge with upstream.

**Comment by rousseldenis** [comment]
@PCatinean Do you want to keep last commit ? If not, could you squash your commits into migration one ?

**Comment by PCatinean** [comment]
@rousseldenis done

**Comment by PCatinean** [comment]
Not sure why commits still show after rebase though

**Comment by rousseldenis** [comment]
> Not sure why commits still show after rebase though

That's normal. They are not present in 12.0 branch. That's the way we do usually in OCA. New repositories are blank ones, without previous commits

**Pull request merged**